rclcpp/include/rclcpp/publisher_factory.hpp

```diff
@@ -75,21 +75,24 @@ struct PublisherFactory
 /// Return a PublisherFactory with functions setup for creating a PublisherT<MessageT, Alloc>.
 template<typename MessageT, typename Alloc, typename PublisherT>
 PublisherFactory
-create_publisher_factory(std::shared_ptr<Alloc> allocator)
+create_publisher_factory(
+  const PublisherEventCallbacks & event_callbacks,
+  std::shared_ptr<Alloc> allocator)
 {
   PublisherFactory factory;
 
   // factory function that creates a MessageT specific PublisherT
   factory.create_typed_publisher =
-    [allocator](
+    [event_callbacks, allocator](
     rclcpp::node_interfaces::NodeBaseInterface * node_base,
     const std::string & topic_name,
     rcl_publisher_options_t & publisher_options) -> std::shared_ptr<PublisherT>
     {
       auto message_alloc = std::make_shared<typename PublisherT::MessageAlloc>(*allocator.get());
       publisher_options.allocator = allocator::get_rcl_allocator<MessageT>(*message_alloc.get());
 
-      return std::make_shared<PublisherT>(node_base, topic_name, publisher_options, message_alloc);
+      return std::make_shared<PublisherT>(node_base, topic_name, publisher_options,
+               event_callbacks, message_alloc);
```

Comment:
I think this indentation is wrong, it should be just two spaces, did `uncrustify` make you do this?

Reply:
I don't remember doing this and I would not do this on purpose. However, `ament_uncrustify` is not catching this. (I tried `ament_uncrustify --reformat src/ros2/rclcpp/`, am I doing something wrong?)

Reply:
Weird, maybe related to the return statement. If `ament_uncrustify --reformat` is ok with it, then that's fine.

Reply:
So it turns out it in fact is `ament_uncrustify` that's making these indentations that are greater than 2 spaces.

Reply:
I have manually removed the extra spaces in the indentation for now.

Reply:
We also suggest to wrap after the open parenthesis (instead of in the middle of arguments). That avoids any attempt to align the following lines and indents them one level.